### PR TITLE
Multipliers for shield generation and hull repair related stats

### PR DESCRIPTION
**Pull request by Amazinite**
Ref: #4407

Adds multipliers for shield generation, hull repair, and their respective energy/fuel/heat stats. 
"Multipliers" are actually additive, with a value of 1 being a 100% increase in the stat, and if you have two outfits with a multiplier of 10%, then the stat will end up being multiplied by 1.2. Negative values are possible, allowing for the reduction in a stat.

### CODE REVIEW

**Comment by Amazinite** [comment]
This needs the ShipInfoDisplay updated to properly reflect the effects of a ship's multipliers on its stats.

**Comment by Amazinite** [comment]
`reverse thrusting energy` is currently the longest attribute name, but 
`shield generation multiplier` would be longer than that, so maybe a different word than multiplier is necessary.
`shield generation booster`? Would still be the longest, but not by as much.

**Comment by Amazinite** [comment]
It might be fine in game as is. Not quite as much space for larger numbers, but I wouldn't expect these attributes to have large numbers in practice. 
image

**Comment by petervdmeer** [comment]
Most other stacked stats in the game use multiplication instead of addition. If one outfit adds 10% shield-boost and another outfit adds 10% shield-boost, then the total boost would be 21% (1.1 * 1.1).

Removing 20% of a stat would be multiplication by 0.8.

Is it critical for this PR that the combination of stats is additive instead of multiplicative? If it is not, then I suggest to switch to multiplicative.

**Comment by Amazinite** [comment]
> Most other stacked stats in the game use multiplication instead of addition.

Really? What stats would those be?
Attributes from various outfits are all added together to get the final value of that attribute for the ship.

**Comment by petervdmeer** [comment]
> > Most other stacked stats in the game use multiplication instead of addition.
> 
> Really? What stats would those be?
> If you're thinking of an attribute like ramscoop, ramscoop is still additive, it's just that the square root of the added value is used to get the diminishing returns effect.

Sorry, I was indeed thinking of stats like `ramscoop`, `outfit scan power` and `cooling inefficiency`, but you are right, they don't get multiplied initially.

> > Is it critical for this PR that the combination of stats is additive instead of multiplicative? If it is not, then I suggest to switch to multiplicative.
> 
> To make various outfits with these stats multiplicatively combine would require adding some way for the game to keep track of which stats are coming from which outfits when installed on a ship instead of just adding everything together.

Yes, it would be a variable that is initialized at 1 and instead of continuously adding the value for every outfit the value would be multiplied for every outfit. Would be a little bit more coding since until now all stats indeed just get added.

> Multiplicative outfits would also be far more difficult for the player to keep track of. (E.g. having 3 10% outfits and 4 5% outfits is easier to add to 50% than it is to figure out that that's actually a 155% boost in some stat, this example also showing the runaway effect of multiplicatively combining outfit attributes.)

Yes, adding is the correct operation if you don't want the effect of one multiplier outfit to also work on top of another multiplier outfit. I just wasn't sure if you indeed intended the multiplier to work that way.

**Comment by Amazinite** [comment]
Ref: #4682
I've hard coded a whitelist for the new attributes added by this PR to allow them to be negative, but #4682 asks that this be something that is listed in the game data. For now the hard coded whitelist should do, though.

File: data/tooltips.txt [hunk 1]
```diff
@@ -134,15 +134,27 @@ tip "hull:"
 tip "hull energy:"
 	`Energy consumed per second when repairing the hull.`
 
+tip "hull energy multiplier:"
+	`A multiplier that affects hull energy. A multiplier of 10% will increase the stat by 10%. Multipliers from multiple outfits are additive.`
```
Comment (petervdmeer) [inline data/tooltips.txt:138]: The tooltip texts here are a bit long and feel a bit "out of lore"; Maybe change to something like:
`Increases the energy consumed when reparing the hull by the given factor.`
This is a little bit less precise, but it seems to match the style of the `hull energy` text better and I think it should be fine for the average player.

Comment (Amazinite) [reply data/tooltips.txt:138]: Well these could potentially be negative (after #4942), so do we want to use the term "increases"?

Comment (petervdmeer) [reply data/tooltips.txt:138]: You could add the text `(Decreases if the factor is negative)` to be complete.

Comment (Amazinite) [reply data/tooltips.txt]: How do they look now?

Comment (petervdmeer) [reply data/tooltips.txt]: Looks good to me.

**Pull request merged**